File: isValid.py
def isValid(s: str) -> bool:
    stack = []
    bracket_mapping = {
        ')': '(',
        ']': '[',
        '}': '{'
    }

    for bracket in s:
        if bracket in bracket_mapping:
            if not stack or stack[-1] != bracket_mapping[bracket]:
                return False
            stack.pop()
        else:
            stack.append(bracket)

    return not stack

s = '()'
print(isValid(s)) # True

s = '({)}'
print(isValid(s)) # False

File: isValid.js
const isValid = (s) => {
    let stack = [];
    let bracketMapping = {
        ')': '(',
        ']': '[',
        '}': '{'
    };

    for(let i = 0; i < s.length; i++) {
        if(s[i] in bracketMapping) {
            if(stack.length === 0 || stack[stack.length - 1] !== bracketMapping[s[i]]) {
                return false;
            }
            stack.pop();
        } else {
            stack.push(s[i]);
        }
    }

    return stack.length === 0;
}

let s = '()';
console.log(isValid(s)); // true

s = '({)}'
console.log(isValid(s)); // false